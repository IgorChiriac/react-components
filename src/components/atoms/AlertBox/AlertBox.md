Styled AlertBox component

```js
import { AlertBox, Link, Text, SizedContainer } from '@zopauk/react-components';

<AlertBox>
  <SizedContainer size="long">
    <Text as="p" mb>
      The personal information we have collected from you will be used to verify your identity.
    </Text>
    <Link href="#">Tell me more in plain English</Link>
  </SizedContainer>
</AlertBox>;
```
