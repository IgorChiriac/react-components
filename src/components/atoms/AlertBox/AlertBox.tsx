import React, { FC } from 'react';
import styled from 'styled-components';
import * as Colors from '../../../constants/colors';
import Alert from '../../icons/Alert/Alert';

export interface IAlertBoxProps extends React.HTMLAttributes<HTMLElement> {
  icon?: FC;
}

const Box = styled.div`
  display: inline-flex;
  padding: 16px;
  border: 1px solid ${Colors.neutral.neutral50};
  border-radius: 8px;
  box-shadow: 0px 2px 6px 0px rgba(0, 0, 0, 0.1);
`;

const IconContainer = styled.div`
  margin-top: 6px;
  margin-right: 16px;
`;

const AlertBox: FC<IAlertBoxProps> = ({ children, className, icon, style }) => (
  <Box className={className} style={style}>
    <IconContainer>{icon || <Alert />}</IconContainer>
    {children}
  </Box>
);

export default AlertBox;
