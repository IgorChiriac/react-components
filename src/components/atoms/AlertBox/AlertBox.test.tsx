import { axe } from 'jest-axe';
import React from 'react';
import { render } from '@testing-library/react';
import AlertBox from './AlertBox';

describe('<AlertBox />', () => {
  it('renders the component with no a11y violations', async () => {
    const { container } = render(<AlertBox />);
    const results = await axe(container.innerHTML);

    expect(container.firstChild).toMatchSnapshot();
    expect(results).toHaveNoViolations();
  });
});
