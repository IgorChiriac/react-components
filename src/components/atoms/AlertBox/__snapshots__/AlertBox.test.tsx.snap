// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertBox /> renders the component with no a11y violations 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 16px;
  border: 1px solid #E2E3E7;
  border-radius: 8px;
  box-shadow: 0px 2px 6px 0px rgba(0,0,0,0.1);
}

.c1 {
  margin-top: 6px;
  margin-right: 16px;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <svg
      height="20px"
      viewBox="0 0 20 20"
      width="20px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M-2-2h24v24H-2z"
        />
        <path
          d="M5.58 2.08L4.15.65C1.75 2.48.17 5.3.03 8.5h2a8.445 8.445 0 0 1 3.55-6.42zM17.97 8.5h2c-.15-3.2-1.73-6.02-4.12-7.85l-1.42 1.43a8.495 8.495 0 0 1 3.54 6.42zM16 9c0-3.07-1.64-5.64-4.5-6.32V2c0-.83-.67-1.5-1.5-1.5S8.5 1.17 8.5 2v.68C5.63 3.36 4 5.92 4 9v5l-2 2v1h16v-1l-2-2V9zm-6 11c.14 0 .27-.01.4-.04.65-.14 1.18-.58 1.44-1.18.1-.24.15-.5.15-.78h-4c.01 1.1.9 2 2.01 2z"
          fill="#A9ACBA"
          fill-rule="nonzero"
        />
      </g>
    </svg>
  </div>
</div>
`;
